tenzir/CMakeLists.txt

```diff
@@ -116,82 +116,104 @@ endif ()
-      "diff" "--exit-code" "--" "*.ref"
-    DEPENDS update-integration
-    COMMENT "Diffing all integration test results"
-    USES_TERMINAL)
+      "${CMAKE_COMMAND}" "-E" "env"
+      "PATH=$<TARGET_FILE_DIR:tenzir::tenzir>:${PATH}" "--" "${TENZIR_UV_PATH}"
+      "run" "${CMAKE_CURRENT_LIST_DIR}/tests/run.py" "-u"
+    DEPENDS tenzir)
 endif ()
 
-include(ProcessorCount)
-ProcessorCount(parallel_level)
-math(EXPR parallel_level "${parallel_level} + 2")
+if (TENZIR_ENABLE_BATS_TESTS)
+  add_custom_target(integration)
+  TenzirDefineUpdateIntegrationTarget(integration OFF)
+  add_custom_target(integration-tenzir)
+  add_dependencies(integration integration-tenzir)
+  TenzirDefineUpdateIntegrationTarget(
+    integration-tenzir "${CMAKE_CURRENT_SOURCE_DIR}/integration/data/reference")
+
+  # This needs a custom definition instead of the one from
+  # TenzirDefineUpdateIntegrationTarget as it requires two calls to properly diff
+  # within the submodule.
+  if (Git_FOUND AND EXISTS "${PROJECT_SOURCE_DIR}/.git")
+    add_custom_target(
+      diff-integration
+      COMMAND "${GIT_EXECUTABLE}" "-C" "${CMAKE_SOURCE_DIR}" "add"
+              "--intent-to-add" "--all"
+      COMMAND "${GIT_EXECUTABLE}" "-C" "${CMAKE_SOURCE_DIR}" "diff" "--exit-code"
+              "--" "*.ref"
+      COMMAND
+        "${GIT_EXECUTABLE}" "-C" "${CMAKE_SOURCE_DIR}/contrib/tenzir-plugins"
+        "diff" "--exit-code" "--" "*.ref"
+      DEPENDS update-integration
+      COMMENT "Diffing all integration test results"
+      USES_TERMINAL)
+  endif ()
+
+  include(ProcessorCount)
+  ProcessorCount(parallel_level)
+  math(EXPR parallel_level "${parallel_level} + 2")
+
+  file(GLOB_RECURSE _suites CONFIGURE_DEPENDS
+       "${CMAKE_CURRENT_SOURCE_DIR}/integration/tests/*.bats")
+  foreach (suite IN LISTS _suites)
+    get_filename_component(bats_suite_name "${suite}" NAME_WE)
+    string(REGEX REPLACE "_" "-" suite_name "${bats_suite_name}")
+    add_custom_target(
+      "integration-tenzir-${suite_name}"
+      COMMAND
+        "${CMAKE_COMMAND}" -E env
+        PATH="$<TARGET_FILE_DIR:tenzir::tenzir>:\$\$PATH:${CMAKE_CURRENT_SOURCE_DIR}/integration/lib/bats/bin"
+        bats "-T" "--jobs" "${parallel_level}" "${suite}"
+      COMMENT "Executing integration test suite ${suite_name}..."
+      USES_TERMINAL)
+    add_dependencies(integration-tenzir-${suite_name} tenzir)
+    add_dependencies(integration-tenzir integration-tenzir-${suite_name})
+    TenzirDefineUpdateIntegrationTarget(
+      integration-tenzir-${suite_name}
+      "${CMAKE_CURRENT_SOURCE_DIR}/integration/data/reference/${bats_suite_name}")
+  endforeach ()
+
+  unset(parallel_level)
 
-file(GLOB_RECURSE _suites CONFIGURE_DEPENDS
-     "${CMAKE_CURRENT_SOURCE_DIR}/integration/tests/*.bats")
-foreach (suite IN LISTS _suites)
-  get_filename_component(bats_suite_name "${suite}" NAME_WE)
-  string(REGEX REPLACE "_" "-" suite_name "${bats_suite_name}")
+  # link the full
```

Comment:
the full what?

Reply:
This is just shifted by two spaces due to a new `if (TENZIR_ENABLE_BAST_TESTS)` to make these optional; not sure what this comment was about tbh, but it wasn't introduced by this PR.

Reply:
Ah, I was referring to the comment here only. My guideposts in CMake.